### DISCUSSION TITLE
Region = lsm for continental computation, Can we set Region.fill(1) for a global computation include both land and ocean

### DISCUSSION THREAD

**Discussion by XINGXING-ZHANG** [WAM2layers/WAM2layers#284] [Q&A] [answered]
![evaporation recycling ratio2020-2022](https://github.com/WAM2layers/WAM2layers/assets/57533177/333b1c97-09d9-4e97-a5cd-a9b125425b18)
Please！！ I got very high recycling ratio value plot as the figure above. If there are mistakes of my computation？ The output results as follow：

print (outsideContribution_Ratio*100,insideContribution_Ratio*100)
0.0 89.67874054166549
print (Isthis100procent*100)
102.81466661625065
print(water_lost_vertically*100, water_lost_north*100, water_lost_south*100)
13.020959308582059 0.014283748414741074 0.0011005998067240058

**Comment by XINGXING-ZHANG** [comment]
It dosen‘t make sense

**Comment by Peter9192** [comment]
Dear @XINGXING-ZHANG, welcome to the WAM2layers community, thanks for reaching out. 

Without more context it is not possible for us to help you. You ask whether there is a mistake in your computations, but you haven't shared any of your computations. Please describe what you're trying to achieve, what you've done, what is different from what you expected, et cetera. Also, it would be nice if you could introduce yourself briefly so we can start to know each other a bit better. See https://github.com/WAM2layers/WAM2layers/discussions/206

Cheers,
Peter

**Reply by XINGXING-ZHANG** [reply]
Dear @Peter9192, much appreciate your kindly reply and the research conducted by your team. I have updated my introduction in #206. 

- I intend to run model in global using the WAM2layersPython-2.4.08 code, which has been adjusted to suit my data. My purpose in doing this is to assess whether the eruption of the Tonga volcano has an impact by statistically analyzing the precipitation/evaporation recycling ratio in different regions. When I set a specific region (Set the values in the matrix corresponding to the region to 1, and set the values for other regions to 0.), whether using backtrack or forwardtrack, I can obtain the desired results (as shown in the figure below). 
<img width="864" alt="图片1" src="https://github.com/WAM2layers/WAM2layers/assets/57533177/7ec8b12c-f14c-4c34-b512-80dc45be541d">

- But, with both land and ocean masks set to 1 for global computation (Set the entire global matrix to 1), the following results have emerged, clearly showing precipitation/evaporation recycling ratios exceeding 50%. This value is too lager compare to the paper below.
![evaporation recycling ratio2020-2022](https://github.com/WAM2layers/WAM2layers/assets/57533177/e51f0589-d166-42a3-9a62-352bec71ef62)
![Figure 2023-11-20 171612](https://github.com/WAM2layers/WAM2layers/assets/57533177/faa8f258-3c0e-4dd6-8040-7428403490d3)



- For instance, in the article "Contrasting roles of interception and transpiration in the hydrological cycle – Part 2: Moisture recycling," the Region is set as continental land, meaning the mask for all land areas is set to 1, while the mask for the ocean is set to 0. The precipitation/evaporation recycling ratio in following results smaller than 0.6 in most continental area.
<img width="736" alt="微信图片_20231120170531" src="https://github.com/WAM2layers/WAM2layers/assets/57533177/094d5819-df6c-471a-b9e6-cdbc0c993be9">

**Reply by Peter9192** [reply]
Dear @XINGXING-ZHANG, thanks for the clarification.

If I understand you correctly, you calculate (in case of precip) the ratio of local precipitation to total precipitation: $\frac{P_{local}}{P_{total}}$, right? And in the continental case $P_{local}$ is what you obtained when you set the source region to match the land surface mask. So far, so good? 

I'm not an expert on moisture recycling like @ruudvdent, but I would expect that if you set the source region to match both the land and sea area (that is, the full globe), then $P_{local}$ equals $P_{total}$ and hence the ratio should equal 1, or 100%. Is that not what you expected?

**Reply by XINGXING-ZHANG** [reply]
Thank you for you reply. But the following situations have occurred：

1. some location E_track/E_total > 1. In principle, it should be less than 1
2. for  whole continental land（E_track + water_lost)/P_total > 1. In principle, it should be less than 1
![Figure 2023-11-23 130917](https://github.com/WAM2layers/WAM2layers/assets/57533177/f90895b9-6429-4607-929b-23eef2ca5b8a)

**Reply by ruudvdent** [reply]
Indeed, you are right for both points. However, it's a bit hard to assess what went wrong without having more details. I read above also that you are using an old version "WAM2layersPython-2.4.08". Please consider moving to this latest version on this GitHub page, which may help to identify what goes wrong.

**Reply by XINGXING-ZHANG** [reply]
@ruudvdent Thank your for kindly reply. I have read the paper your have published in 2017 show as follows (The residence time of water in the atmosphere revisited). I guess we cannot set the source region include both continental land and ocean (in other words set the global area as source region). That’s maybe why you performed 4 experiments in follows paper. You tracked continental and ocean evaporation, respectively. Instead of directly tracking Continental and Ocean once. Please let me know if my understanding is correct.
<img width="1000" alt="微信图片_20231129162411" src="https://github.com/WAM2layers/WAM2layers/assets/57533177/a4a77439-d1f1-472e-ab0e-e9dc2ff476b2">

**Reply by ruudvdent** [reply, marked as answer]
Dear @XINGXING-ZHANG . I did this separately, because tracking all water yielded to many water losses due to non-water balance closures. With perfect data and a perfect model, tracking all water should have been fine.

**Reply by XINGXING-ZHANG** [reply]
> Dear @XINGXING-ZHANG, welcome to the WAM2layers community, thanks for reaching out.
> 
> Without more context it is not possible for us to help you. You ask whether there is a mistake in your computations, but you haven't shared any of your computations. Please describe what you're trying to achieve, what you've done, what is different from what you expected, et cetera. Also, it would be nice if you could introduce yourself briefly so we can start to know each other a bit better. See #206
> 
> Cheers, Peter
@Peter9192 
Dear Peter,

I hope you are doing well. I want to thank you for your continuous help in understanding and running the models. I have written a paper, and I believe your contributions should be highlighted in it.

I tried sending the draft to you via email, but it was blocked by your email system. My email address is [zhangxx@igsnrr.ac.cn]. If convenient, we can connect via email.

Thank you once again, and I look forward to your feedback.
XINGXING ZHANG

**Reply by Peter9192** [reply]
Dear @XINGXING-ZHANG,

Thank you for the notification. Ruud has forwarded your email to me. I will get back to you.

Cheers,
Peter
